modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataDefinitionResourceImpl.java

```diff
@@ -1221,6 +1225,56 @@ private void _removeFieldsFromDataListViews(
 		}
 	}
 
+	private void _reorderNestedDDMFormFields(DDMForm ddmForm) throws Exception {
+		for (DDMFormField ddmFormField : ddmForm.getDDMFormFields()) {
+			if (!StringUtil.equals(
+					ddmFormField.getType(),
+					DDMFormFieldTypeConstants.FIELDSET)) {
+
+				continue;
+			}
```

Comment:
@marcelabc what happens if we have a `fieldSet` inside another `fieldSet` and we change the fields order in the second level `fieldSet`? I'm asking because you are calling `_reorderNestedDDMFormFields` only for the top level fields.